Tests/Rules/UseCompatibleTypes.Tests.ps1

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Import-Module "$PSScriptRoot/../../out/PSScriptAnalyzer"
+
+$script:RuleName = 'UseCompatibleTypes'
+$script:AnyProfileConfigKey = 'AnyProfilePath'
+$script:TargetProfileConfigKey = 'TargetProfilePaths'
+
+$script:Srv2012_3_profile = 'win-8_x64_6.2.9200.0_3.0_x64_4.0.30319.42000_framework'
+$script:Srv2012r2_4_profile = 'win-8_x64_6.3.9600.0_4.0_x64_4.0.30319.42000_framework'
+$script:Srv2019_5_profile = 'win-8_x64_10.0.17763.0_5.1.17763.134_x64_4.0.30319.42000_framework'
+$script:Srv2019_6_1_profile = 'win-8_x64_10.0.17763.0_6.1.2_x64_4.0.30319.42000_core'
+$script:Ubuntu1804_6_1_profile = 'ubuntu_x64_18.04_6.1.2_x64_4.0.30319.42000_core'
+
+$script:TypeCompatibilityTestCases = @(

```

Comment:
I need some inspiration as to what else to test here. I'm looking for .NET types that are conspicuously missing in various PowerShell versions.

I'm also aware that this rule currently can't take account of an `Add-Type` introducing a type and for some reason `System.Drawing.Color` is showing up on some platforms are not being there...